src/core/device.c

```diff
@@ -579,10 +579,12 @@ static int device_process_new(Manager *m, struct udev_device *dev) {
                 if (r < 0)
                         return log_warning_errno(r, "Failed to add parse SYSTEMD_ALIAS for %s: %m", sysfs);
 
-                if (path_is_absolute(word))
-                        (void) device_setup_unit(m, dev, word, false);
-                else
+                if (!path_is_absolute(word))
                         log_warning("SYSTEMD_ALIAS for %s is not an absolute path, ignoring: %s", sysfs, word);
+                else if (!path_is_normalized(word))
+                        log_warning("SYSTEMD_ALIAS for %s is not a normalized path, ignoring: %s", sysfs, word);
```

Comment:
Hm, that seems unnecessarily harsh. If I'm reading the code correctly, we would accept such paths and escape them, so this change breaks backward compat. And people tend to autogenerate rules, so a non-normalized path is possible.

Reply:
this is not about deps, this is about adding alias names for units. For deps defined through udev rules we are tolerant, and will mangle. But I am sure if we accept device aliases we should be be very strict, as otherwise we might get into quite some ambiguity troubles